src/python-common/CMakeLists.txt

```diff
@@ -1,3 +1,6 @@
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/ceph/version.py.in
+  ${CMAKE_CURRENT_SOURCE_DIR}/ceph/version.py @ONLY)
```

Comment:
please do not create source file in `src`. put them in `${CMAKE_CURRENT_BINARY_DIR}` instead.

Reply:
This requirement is killing this whole approach unfortunately. Let me think of an alternative approach.

Reply:
@tchaikov do you think the same applies to https://github.com/ceph/ceph/pull/41855?